v2/cf-nodejs-logging-support/src/lib/config/config.ts

```diff
@@ -0,0 +1,143 @@
+import { MergedConfig, ConfigObject, ConfigField, customFieldsFormat } from './interfaces';
+import * as coreConfig from './config-core.json';
+import * as requestConfig from './config-request.json';
+import * as cfConfig from './config-cf.json';
+import * as kymaConfig from './config-kyma.json';
+import * as appLoggingConfig from './config-app-logging.json';
+import * as cloudLoggingConfig from './config-cloud-logging.json';
+import EnvService from '../core/env-service';
+
+export default class Config {
+
+    private static instance: Config;
+
+    private config: MergedConfig = {
+        "fields": [],
+        "customFieldsFormat": "cloud-logging",
+        "outputStartupMsg": false
+    }
+
+    private constructor() {}
+
+    public static getInstance(): Config {
+        if (!Config.instance) {
+            let env = EnvService.getRuntimeName();
+            let boundServices = EnvService.getBoundServices();
+            let envConfig = () => {
+                if (env == "Kyma") {
+                    return kymaConfig as ConfigObject;
+                }
+                return cfConfig as ConfigObject;
+            }
+            let boundServiceConfig = () => {
+                if (boundServices["application-logging"]) {
+                    return appLoggingConfig as ConfigObject;
+                }
+
+                return cloudLoggingConfig as ConfigObject;
+            }
+
+            Config.instance = new Config();
+
+            Config.instance.addConfig([
+                coreConfig as ConfigObject,
+                requestConfig as ConfigObject,
+                envConfig(),
+                boundServiceConfig()
```

Comment:
could be a list that is applied here, instead of predefined entries, so we have the future possibility of adding configs more easily here.